protobuf/source_reflection.go

```diff
@@ -0,0 +1,71 @@
+package protobuf
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/grpcreflect"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	rpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/resolver/manual"
+)
+
+var (
+	// ErrorCouldNotDialReflectionServer is an error for when the reflection server could not be reached.
+	ErrorCouldNotDialReflectionServer = errors.New("could not reach reflection server")
+)
+
+// ReflectionArgs are args for constructing a DescriptorProvider that reaches out to a reflection server.
+type ReflectionArgs struct {
+	Caller  string
+	Service string
+	Peers   []string
+	Timeout time.Duration
+}
+
+// NewDescriptorProviderReflection returns a DescriptorProvider DescriptorProvider that reaches out to
+// a reflection server to access filedescriptors.
```

Comment:
nit: `filedescriptors` -> `file descriptors`.

Reply:
nit @amckinney `FileDescriptorSet`?

Reply:
I'll make it file descriptors, as technically according to the api you get multiple of those rather than one set.